include/qunit.hpp

```diff
@@ -326,6 +328,10 @@ class QUnit : public QInterface {
     bool INTSCOptimize(
         bitCapInt toMod, bitLenInt start, bitLenInt length, bool isAdd, bitLenInt carryIndex, bitLenInt overflowIndex);
 
+    template <typename F>
```

Comment:
Unlike the other ones, this may not need to be a template because the signature of the passed in method doesn't ever change.

Reply:
I see your point, and I would prefer not to use the template. I tried to replace it with a function pointer. However, that doesn't work easily, because the lambda needs to capture `this` to dispatch a QUnit member instance method, (as by `[&]`). The lambda needs to capture to access the instance property, and lambdas with capture can't be cast to function pointers without using std::function. This comes at a performance cost, (probably slight, in our particular case,) but it's also significantly harder to implement and read. It makes me question what the template needs to do to handle the typing of the lambda with capture, but the overhead is probably still mostly at compile time, and it's simpler code.